README.md

```diff
@@ -597,6 +767,122 @@ And the only place it's recommended to use `cleanupProps()` is as the last step
 
 
 
+### `declarePropTypesNarrowing()`
+```js
+declarePropTypesNarrowing: () => Function
+```
+
+When "you know better than Typescript" that a given existing prop type can be refined/narrowed, you can use `declarePropTypesNarrowing()`
+to instruct Typescript about the prop types. This could come in handy eg after a `branch()`
+(though prefer [`branchIfNullish()`](#branchifnullish)/[`branchIfFalsy()`](#branchiffalsy)/[`branchIfEmpty()`](#branchifempty) if they
+cover your branch condition, since they do the prop type-narrowing for you):
+
+```typescript
+import {declarePropTypesNarrowing} from 'ad-hok-utils'
+
+interface Item {
+  id: string
+  name?: string
+}
+
+const MyComponent: FC<{item: Item}> = flowMax(
+  branch(({item: {name}}) => !name, renderNothing()),
+  declarePropTypesNarrowing<{item: Item & {name: string}}>(),
```

Comment:
Maybe use an example that `branchIfNullish` or the other built-in helpers can't handle? eg. a prop that can be either a string or a number, branching if it's a number.